src/hotspot/share/opto/loopUnswitch.cpp

```diff
@@ -31,31 +31,30 @@
 #include "opto/predicates.hpp"
 #include "opto/rootnode.hpp"
 
-//================= Loop Unswitching =====================
+// Loop Unswitching is a loop optimization to move an invariant, non-loop-exiting test in the loop body before the loop.
+// This is done by duplicating the loop and placing the if-path in one unswitched loop version (i.e. the true-path-loop)
+// and the else-path in the other (i.e. the false-path-loop). The invariant test only needs to be checked once and
+// depending on the outcome, one or the other unswitched loop version is executed:
```

Comment:
> BTW: why do we not hoist exit conditions also? Would that not also be profitable? Just out of curiosity...

This can better be done with a simple loop peeling which can hoist the invariant checks. Also see:
https://github.com/openjdk/jdk/blob/0c2def0e3eaa4925746158294f798d37d42c50f3/src/hotspot/share/opto/loopTransform.cpp#L499-L502

Reply:
Ah yes, makes sense!
Maybe add a small comment about loop peeling then? Just because you do mention that we don't handle exits, that automatically raises that question.